adapter/internal/oasparser/model/api_yaml.go

```diff
@@ -22,11 +22,104 @@ import (
+		Context                    string   `json:"context,omitempty"`
+		Version                    string   `json:"version,omitempty"`
+		RevisionID                 int      `json:"revisionId,omitempty"`
+		APIType                    string   `json:"type,omitempty"`
+		LifeCycleStatus            string   `json:"lifeCycleStatus,omitempty"`
+		EndpointImplementationType string   `json:"endpointImplementationType,omitempty"`
+		AuthorizationHeader        string   `json:"authorizationHeader,omitempty"`
+		SecurityScheme             []string `json:"securityScheme,omitempty"`
+		OrganizationID             string   `json:"organizationId,omitempty"`
+		EndpointConfig             struct {
+			EndpointType                 string              `json:"endpoint_type,omitempty"`
+			LoadBalanceAlgo              string              `json:"algoCombo,omitempty"`
+			LoadBalanceSessionManagement string              `json:"sessionManagement,omitempty"`
+			LoadBalanceSessionTimeOut    string              `json:"sessionTimeOut,omitempty"`
+			APIEndpointSecurity          APIEndpointSecurity `json:"endpoint_security,omitempty"`
+			RawProdEndpoints             interface{}         `json:"production_endpoints,omitempty"`
+			ProductionEndpoints          []EndpointInfo
+			ProductionFailoverEndpoints  []EndpointInfo `json:"production_failovers,omitempty"`
+			RawSandboxEndpoints          interface{}    `json:"sandbox_endpoints,omitempty"`
+			SandBoxEndpoints             []EndpointInfo
+			SandboxFailoverEndpoints     []EndpointInfo `json:"sandbox_failovers,omitempty"`
+			ImplementationStatus         string         `json:"implementation_status,omitempty"`
+		} `json:"endpointConfig,omitempty"`
+		Operations []OperationYaml `json:"Operations,omitempty"`
+	} `json:"data"`
+}
+
+// APIEndpointSecurity represents the structure of endpoint_security param in api.yaml
+type APIEndpointSecurity struct {
+	Production EndpointSecurity `json:"production,omitempty"`
+	Sandbox    EndpointSecurity `json:"sandbox,omitempty"`
+}
+
+// EndpointSecurity contains parameters of endpoint security at api.json
+type EndpointSecurity struct {
+	Password         string            `json:"password,omitempty" mapstructure:"password"`
+	Type             string            `json:"type,omitempty" mapstructure:"type"`
+	Enabled          bool              `json:"enabled,omitempty" mapstructure:"enabled"`
+	Username         string            `json:"username,omitempty" mapstructure:"username"`
+	CustomParameters map[string]string `json:"customparameters,omitempty" mapstructure:"customparameters"`
+}
+
+// EndpointInfo holds config values regards to the endpoint
+type EndpointInfo struct {
+	Endpoint string `json:"url,omitempty"`
+	Config   struct {
+		ActionDuration string `json:"actionDuration,omitempty"`
+		RetryTimeOut   string `json:"retryTimeOut,omitempty"`
+	} `json:"config,omitempty"`
+}
+
+// OperationYaml holds attributes of APIM operations
+type OperationYaml struct {
```

Comment:
Should this be `APIOperation`???

Reply:
This is the object to which the Operation in the api.yaml directly gets unmarshelled. I feel keeping this as OperationYaml would give an hint that this is related to api.yaml. We already have an `Operation` struct and it is located in `api_operation.go`. Therefore, keeping this way would avoid confusions due to that file name too.